adopt-github-release/src/main/groovy/net/adoptopenjdk/release/UploadFiles.groovy

```diff
@@ -0,0 +1,116 @@
+package net.adoptopenjdk.release
+
+import groovy.cli.picocli.CliBuilder
+import groovy.cli.picocli.OptionAccessor
+import groovy.transform.CompileStatic
+import org.kohsuke.github.GHAsset
+import org.kohsuke.github.GHRelease
+import org.kohsuke.github.GHRepository
+import org.kohsuke.github.GitHub
+
+import java.nio.file.Files
+
+@CompileStatic
+class UploadAdoptReleaseFiles {
+
+    private final String tag
+    private final String description
+    private final boolean release
+    private final List<File> files
+    private final String version
+
+    UploadAdoptReleaseFiles(String tag, String description, boolean release, String version, List<File> files) {
+        this.tag = tag
+        this.description = description
+        this.release = release
+        this.files = files
+        this.version = version
+    }
+
+    void release() {
+        GHRepository repo = getRepo()
+        GHRelease release = getRelease(repo)
+        uploadFiles(release)
+    }
+
+    private GHRepository getRepo() {
+        String token = System.getenv("GITHUB_TOKEN")
+        if (token == null) {
+            System.err.println("Could not find GITHUB_TOKEN")
+            System.exit(1)
+        }
+
+        GitHub github = GitHub.connectUsingOAuth(token)
+        return github.getRepository("AdoptOpenJDK/open${version}-binaries")
+    }
+
+    private void uploadFiles(GHRelease release) {
+        List<GHAsset> assets = release.getAssets()
+        files.each { file ->
+            // Delete existing asset
+            assets
+                    .find({ it.name == file.name })
+                    .each { GHAsset existing ->
+                println("Updating ${existing.name}")
```

Comment:
Not really updating, but deleting right?

Reply:
It is an update, to avoid errors uploading duplicate assets I am deleting the existing asset first, so it is an update